CTFd/api/v1/challenges.py

```diff
@@ -121,55 +161,51 @@ def get(self, query_args):
         field = str(query_args.pop("field", None))
         filters = build_model_filters(model=Challenges, query=q, field=field)
 
-        # This can return None (unauth) if visibility is set to public
-        user = get_current_user()
-
-        # Admins can request to see everything
-        if is_admin() and request.args.get("view") == "admin":
-            challenges = (
-                Challenges.query.filter_by(**query_args)
-                .filter(*filters)
-                .order_by(Challenges.value)
-                .all()
-            )
-            solve_ids = set([challenge.id for challenge in challenges])
+        # Admins get a shortcut to see all challenges despite pre-requisites
+        admin_view = is_admin() and request.args.get("view") == "admin"
+
+        solve_counts, user_solves = dict(), set()
+        if scores_visible() and accounts_visible():
+            solve_count_dfl = 0
+            # Build a query for to show challenge solve information. We only
+            # give an admin view if the request argument has been provided.
+            #
+            # NOTE: This is different behaviour to the challenge detail
+            # endpoint which only needs the current user to be an admin rather
+            # than also also having to provide `view=admin` as a query arg.
+            solves_q = _build_solves_query(admin_view=admin_view)
+            # Aggregate the query results into the hashes defined at the top of
+            # this block for later use
+            for chal_id, solve_count, solved_by_user in solves_q:
+                solve_counts[chal_id] = solve_count
+                if solved_by_user:
+                    user_solves.add(chal_id)
         else:
-            challenges = (
-                Challenges.query.filter(
-                    and_(Challenges.state != "hidden", Challenges.state != "locked")
-                )
-                .filter_by(**query_args)
-                .filter(*filters)
-                .order_by(Challenges.value)
-                .all()
+            # This is necessary to match the challenge detail API which returns
+            # `None` for the solve count if visiblity checks fail
+            solve_count_dfl = None
+
+        # Build the query for the challenges which may be listed
+        chal_q = Challenges.query
+        # Admins can see hidden and locked challenges in the admin view
+        if not admin_view:
```

Comment:
nit: I prefer these as `if admin_view is False`
```suggestion
        if admin_view is False:
```